### ISSUE TITLE
User scope override settings not working as expected for compilers.yaml

### ISSUE THREAD

**Issue by frankwillmore** [spack/spack#11149]
I'm attempting to override system settings with per-user settings. In particular, I am unable to override a compiler specification, and observe the following collision when attempting to spec any package using the compiler definition:

```
willmore@thetalogin1:~$ spack spec tmux%intel@18.0.0.128
Input spec
--------------------------------
tmux%intel@18.0.0.128

Concretized
--------------------------------
==> Error: Compiler configuration contains entries with duplicate specification (intel@18.0.0.128, cray-cnl8-mic_knl) in the following files:
        /lus/theta-fs0/software/spack/spack-dev/etc/spack/compilers.yaml: 1 time
        /home/willmore/.spack/cray/compilers.yaml: 1 time

The intent is to be able to use a user-defined compiler to override a system version as described in the spack documentation here: 
https://spack.readthedocs.io/en/latest/configuration.html#configuration-scopes
```

### Information on your system

Cray XC-40 (ALCF-theta)
```
~/.spack/compilers.yaml
$spack/etc/spack/compilers.yaml
```

**Comment by scheibelp** [comment]
Assigning @becker33 because his PR at https://github.com/spack/spack/pull/11910 should resolve this.

**Comment by tgamblin** [comment]
Fixed by #11910.  Please reopen if this is still an issue.
